### PR TITLE
No object literals in props

### PR DESCRIPTION
**Pull request by manzoorwanijk**
This PR memoizes all the object literal props to avoid re-renders

### CODE REVIEW

File: assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx [hunk 1]
```diff
@@ -2,23 +2,28 @@ import React, { forwardRef } from 'react';
...
-));
+const Trigger = forwardRef<typeof IconButton, TriggerProps>(({ tooltip, ...props }, ref) => {
+	const tooltipProps = useMemoStringify<TooltipProps>({ placement: 'top' });
+	return (
+		<IconButton
+			{...props}
+			borderless
+			color='white'
+			className='ee-timezone-info__button'
+			icon={() => <GlobalOutlined noMargin size='small' />}
```
Comment (tn3rb) [inline assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx:21]: what about callbacks like this?
they can be wrapped in `useCallback` no?

we can do that in another PR though

File: assets/src/domain/eventEditor/ui/datetimes/datesList/filterBar/controls/SalesControl.tsx [hunk 2]
```diff
@@ -5,39 +5,34 @@ import { SelectInput } from '@appInputs/SelectInput';
...
+		{
+			value: DatetimeSales.above50Capacity,
+			label: __('dates above 50% capacity'),
+		},
+		{
+			value: DatetimeSales.below50Capacity,
+			label: __('dates below 50% capacity'),
+		},
+	]);
+	return <SelectInput label={__('sales')} value={sales} options={options} onChangeValue={setSales} />;
```
Comment (tn3rb) [inline assets/src/domain/eventEditor/ui/datetimes/datesList/filterBar/controls/SalesControl.tsx:35]: NOICE

File: assets/src/application/ui/display/EmptyState.tsx [hunk 3]
```diff
@@ -12,11 +13,11 @@ interface EmptyStateProps {
 
 const EmptyState: React.FC<EmptyStateProps> = ({ children, description, title, ...props }) => {
 	const className = classNames(props.className, 'ee-empty-state');
-	const iconProps = {
+	const iconProps = useMemoStringify({
 		color: 'gray.100',
 		name: 'question-outline',
 		size: '96px',
-	};
+	});
```
Comment (nicolad) [inline assets/src/application/ui/display/EmptyState.tsx:20]: this was already extracted wouldn't be simpler if we'd pull this out of component ?

File: assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx (hunk 1 above)
Comment (nicolad) [reply assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx:21]: or even better we might update `IconButton` such that it receives just JSX;
I think I read in one of KCD articles that passing JSX from the parent is more performant than rendering that at child level

File: assets/src/infrastructure/ui/inputs/dateTime/Datepicker/index.tsx [hunk 4]
```diff
@@ -22,9 +23,11 @@ const Datepicker: React.FC<DatepickerProps> = ({ value, onChange, onChangeValue,
 		[onChange, onChangeValue]
 	);
 
+	const dateValue = useMemoStringify([value]);
+
 	return (
 		<div className={className}>
-			<BaseUIDatepicker {...props} onChange={onChangeHandler} value={[value]} />
+			<BaseUIDatepicker {...props} onChange={onChangeHandler} value={dateValue} />
```
Comment (nicolad) [inline assets/src/infrastructure/ui/inputs/dateTime/Datepicker/index.tsx:30]: good one :+1:

File: assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/StatusControl.tsx [hunk 5]
```diff
@@ -5,51 +5,49 @@ import { SelectInput } from '@appInputs/SelectInput';
...
+			{
+				value: TicketsStatus.expiredOnly,
+				label: __('expired tickets only'),
+			},
+			{
+				value: TicketsStatus.trashedOnly,
+				label: __('trashed tickets only'),
+			},
+		],
+		[isChained]
```
Comment (nicolad) [inline assets/src/domain/eventEditor/ui/tickets/ticketsList/filterBar/controls/StatusControl.tsx:48]: nice

File: assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply assets/src/application/ui/display/TimezoneTimeInfo/Trigger.tsx:21]: No they cannot be wrapped in `useCallback` because `icon` expects a React Component. May be `React.memo` instead.
